src/fields/MultiSelect.tsx

```diff
@@ -65,6 +69,24 @@ export function MultiSelect<OptionValue extends OptionValueType = string>({
     () => (value || []).map(valueItem => getRsuiteValueFromOptionValue(valueItem, optionValueKey)),
     [optionValueKey, value]
   )
+  const searchBy = useMemo(
+    () =>
+      customSearchRef.current
+        ? (query: string, _, item: OptionAsRsuiteItemDataType<OptionValue>) => {
+            if (!customSearchRef.current || !query.trim().length) {
+              return true
+            }
+
+            if (query !== lastSearchQueryRef.current) {
+              lastSearchQueryRef.current = query
+              customSearchLabelMatchesRef.current = customSearchRef.current.find(query).map(option => option.label)
```

Comment:
Can we have a callback to have only `find` and not the `customSearchRef` ?

Reply:
Pas possible en l'état actuel je pense, ça aurait été résolu en passant par la prop `onSearch` mais elle pose des problèmes de synchronization avec l'appel de la fonction `searchBy`.